torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1497,6 +1507,41 @@ void ProcessGroupNCCL::watchdogHandler() {
     dumpPipe.emplace(rank_);
   }
   while (!done || !terminateProcessGroup_.load()) {
+    std::unordered_set<std::string> allCommIds;
+    {
+        std::lock_guard<std::mutex> lock(mutex_);
+        for (auto it = devNCCLCommMap_.begin(); it != devNCCLCommMap_.end(); it++) {
+          auto& ncclComms = it->second;
+          for (const auto& ncclComm : ncclComms) {
+              allCommIds.emplace(buildNcclUniqueIdStr(ncclComm->getNcclId()));
+          }
+        }
+    }
+    // Abort communicators that have been aborted on other ranks
+    // Otherwise the other ranks may hang indefinitely
+    for (const auto& commId : allCommIds) {
+      // If not already aborted
+      if (abortedComms_.find(commId) == abortedComms_.end()) {
+        const auto& storeKey = getNcclAbortedCommStoreKey(commId);
```

Comment:
i'm a bit wary of adding burden to the TCPStore here, it could cause scalability issues if we're not careful.  And also generally wary of adding complexity to the watchdog loops.  Would rather be simplifying them.

Hoping we can find another solution here.